src/askContainer/commands/viewAllSkills.ts

```diff
@@ -146,6 +150,42 @@ export class ViewAllSkillsCommand extends AbstractCommand<void> {
         }
     }
 
+    private async checkAndSetQpItems(
+        context: vscode.ExtensionContext,
+        qp: vscode.QuickPick<vscode.QuickPickItem>
+    ): Promise<void> {
+        Logger.verbose(`Calling method: ${this.commandName}.checkAndSetQpItems, args:`, qp);
+        const previous = context.globalState.get(VIEW_ALL_SKILLS_LAST_UPDATE_TIME) as number;
+        const now = new Date().getTime();
+
+        if (previous === undefined || (previous !== undefined && this.calculateTimeExceedHourly(previous, now, NUMBER_OF_INTERVALS))) {
+            await this.setLastUpdateTimeAndRefreshList(context, now, qp);
+            return;
+        }
+        await this.setQpItems(context, qp);
+    }
+
+    private calculateTimeExceedHourly(previous: number, now: number, limit: number) {
+        Logger.verbose(`Calling method: ${this.commandName}.calculateTimeExceedHourly, args:`, previous, now, limit);
+        return Math.abs(now - previous) / HOUR >= limit;
+    }
+
+    private async setLastUpdateTimeAndRefreshList(context: vscode.ExtensionContext, now: number, qp: vscode.QuickPick<vscode.QuickPickItem>) {
+        Logger.verbose(`Calling method: ${this.commandName}.setLastUpdateTimeAndRefreshList, args:`, now, qp);
+        void context.globalState.update(VIEW_ALL_SKILLS_LAST_UPDATE_TIME, now);
+        clearCachedSkills(context);
+        await this.setQpItems(context, qp);
+    }
+
+    private async checkSkillExist(context: CommandContext, skillInfo: SmapiResource<SkillInfo>): Promise<void> {
+        Logger.verbose(`Calling method: ${this.commandName}.checkSkillExist`);
+        try {
+            await getSkillMetadata(skillInfo.data.skillSummary.skillId!, SKILL.STAGE.DEVELOPMENT, context.extensionContext);
+        } catch (error) {
+            throw error;
+        }
```

Comment:
nit: Why do you need this `catch` block if you are just throwing it back to the caller?

Reply:
Rmoved `checkSkillExist` method, but call `getSkillMetadata` directly.